pkg/sql/sem/tree/expr.go

```diff
@@ -1626,7 +1626,11 @@ func (node *CastExpr) Format(ctx *FmtCtx) {
 		if _, ok := node.Expr.(*StrVal); ok {
 			ctx.FormatTypeReference(node.Type)
 			ctx.WriteByte(' ')
-			ctx.FormatNode(node.Expr)
+			if ctx.HasFlags(FmtHideConstants) {
```

Comment:
i think this might need an explanatory comment.

could you add a comment saying that "we need to replace this with a quoted string constant because the grammar requires a string constant rather than an expression for this form of casting in the `typed_literal` rule"

https://github.com/cockroachdb/cockroach/blob/6eafde99ef6bf3162da1fc59fd5c12ca4ac5fb9e/pkg/sql/parser/sql.y#L11444